internal/operator-controller/rukpak/bundle/config.go

```diff
@@ -6,119 +6,498 @@ import (
+//
+// Note: nil config is treated as {} and validated against the schema. This means
+// nil configs will error if the bundle requires fields (e.g., OwnNamespace-only bundles
+// require watchNamespace). Callers in applier/provider.go already convert nil to {}
+// before calling this function, so this behavior is consistent.
+func UnmarshalConfig(bytes []byte, schemaProvider ConfigSchemaProvider, installNamespace string) (*Config, error) {
+	// nil config becomes {} so we can validate required fields
 	if bytes == nil {
-		return nil, nil
+		bytes = []byte("{}")
 	}
 
-	bundleConfig := &Config{}
-	if err := yaml.UnmarshalStrict(bytes, bundleConfig); err != nil {
-		return nil, fmt.Errorf("error unmarshalling registry+v1 configuration: %w", formatUnmarshalError(err))
+	// Step 1: Get the schema from the bundle
+	schema, err := schemaProvider.GetConfigSchema()
+	if err != nil {
+		return nil, fmt.Errorf("error getting configuration schema: %w", err)
 	}
 
-	// collect bundle install modes
-	bundleInstallModeSet := sets.New(rv1.CSV.Spec.InstallModes...)
+	// Step 2: Validate against the schema
+	if err := validateConfigWithSchema(bytes, schema, installNamespace); err != nil {
+		return nil, fmt.Errorf("error validating configuration: %w", err)
+	}
 
-	if err := validateConfig(bundleConfig, installNamespace, bundleInstallModeSet); err != nil {
-		return nil, fmt.Errorf("error unmarshalling registry+v1 configuration: %w", err)
+	// Step 3: Parse into Config struct
+	// We use yaml.Unmarshal (non-strict) instead of yaml.UnmarshalStrict because schema validation already
+	// ensures correctness. For schemas that allow additional properties (e.g., Helm charts), extra fields
+	// beyond watchNamespace will pass validation but won't be stored in Config. For registry+v1 bundles,
+	// additionalProperties: false rejects extra fields during schema validation.
+	bundleConfig := &Config{}
+	if err := yaml.Unmarshal(bytes, bundleConfig); err != nil {
+		return nil, fmt.Errorf("error unmarshalling configuration: %w", formatUnmarshalError(err))
 	}
 
 	return bundleConfig, nil
 }
 
-// validateConfig validates a *bundle.Config against the bundle's supported install modes and the user-give installNamespace.
-func validateConfig(config *Config, installNamespace string, bundleInstallModeSet sets.Set[v1alpha1.InstallMode]) error {
-	// no config, no problem
-	if config == nil {
-		return nil
+// buildSchemaForInstallModes creates validation rules for registry+v1 bundles.
+//
+//   - AllNamespaces only -> watchNamespace not allowed (watches all namespaces)
+//   - OwnNamespace only -> watchNamespace required (must equal install namespace)
+//   - SingleNamespace only -> watchNamespace required (must NOT equal install namespace)
+//   - AllNamespaces + OwnNamespace -> watchNamespace optional
+//   - Empty install modes -> no configuration allowed (bundle is invalid)
+func buildSchemaForInstallModes(installModes sets.Set[v1alpha1.InstallMode]) (string, error) {
```

Comment:
In the not-so-distant future we'll introduce subscription config fields, so maybe a more generic name like:

```suggestion
func buildBundleConfigSchema(installModes sets.Set[v1alpha1.InstallMode]) (string, error) {
```